assignments/arrays.js

```diff
@@ -63,34 +63,131 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
+//     for(let i=0; i<arr.length; i++){
+//         carModels.push(arr[i].car_make);
+//     };
+//     retrun carModels; 
+// }
+
+// const modelSort = (arr) => {
+//         let modelList = [];
+//         for(let i = 0; i<arr; i++){
+//             modelList.push(arr[i].car_model);
+//         }
+//         return modelList;
+//     };
+//  
+// carModels = modelSort(inventory);
+
+// let carModels =[];
+// for( let i=0; i<arr.length; i++){
+//     for (prop in arr[i]){
+//         carModels.push(arr[i].car_model)
+//     };
+// }
+
+
+
+// let carModels = inventory.sort((a,b)=>{
+//     let modelA = a.car_model.toLowerCase();
+//     let modelB = b.car_model.toLowerCase();
+//     if(modelA < modelB){
+//         return -1; 
+//     }
+// });
 let carModels = [];
-console.log();
+
+const sortModels = (arr) => {
+    for(let i = arr.length -1; i >= 0; i--){
+        for(let j = 1; j<=i; j++){
+            if(arr[j-1].car_model>arr[j].car_model){
+                let temp = arr[j-1];
+                arr[j-1] = arr[j];
+                arr[j] = temp;
+            }
+        }
+    }
+    return arr;
+}
+
+// console.log(sortModels(inventory));
+
+
+
+
 
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
 let carYears = [];
-console.log();
+
+const findYears = (arr) => {
+    let yearArr = []
+    for(let i = 0; i < arr.length; i++){
+        yearArr.push(arr[i].car_year);
+    }
+    return yearArr;
+}
+
+carYears = findYears(inventory);
+// console.log(carYears);
 
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
 let oldCars =[];
-console.log(); 
+
+const sortByAge = (arr, modelYear) => {
+    let arr2 = [];
+    for(let i = 0; i < arr.length; i++){
+        if(modelYear > arr[i].car_year){
+            arr2.push(arr[i]);
+        }
+    }
+    return arr2;
+}
+oldCars = sortByAge(inventory, 2000);
+// console.log(oldCars); 
 
 // ==== Challenge 6 ====
 // A buyer is interested in seeing only BMW and Audi cars within the inventory.  Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, use JSON.stringify() to show the results of the array in the console.
 let BMWAndAudi =[];
-console.log();
+
+const makeSearch = (arr, make1, make2) => {
+    let arr2 = [];
+    for(let i = 0; i < arr.length; i++){
+        if(arr[i].car_make.toLowerCase() === make1.toLowerCase()){
+            arr2.push(arr[i]);
+        } else if(arr[i].car_make.toLowerCase() === make2.toLowerCase()) {
+            arr2.push(arr[i]);
+        }
+    }
+    return arr2;
+}
+
+BMWAndAudi = makeSearch(inventory, 'BMW', "AUDI");
+console.log(BMWAndAudi);
```

Comment:
well done and good job making use of `.toLowerCase()` to account for an edge case of different case conventions, but remember you need to `JSON.stringify()` the final result per the instructions